### PR TITLE
Enchanced a function in pci.py program_v2

### PR DESCRIPTION
**Pull request by vrbagalkote**
Enchanced the function to return the interface of all types
Earlier it was only valid for nic, now it will work for ant type

Signed-off-by: Venkat R B <vrbagal1@linux.vnet.ibm.com>

### CODE REVIEW

File: avocado/utils/pci.py [hunk 1]
```diff
@@ -97,20 +98,25 @@ def get_disks_in_pci_address(pci_address):
     return disk_list
 
 
-def get_nics_in_pci_address(pci_address):
+def get_interface_in_pci_address(pci_address, pci_class):
```
Comment (ldoktor) [inline avocado/utils/pci.py]: Please keep the old `get_nics_in_pci_address` function pointing to `get_interface_in_pci_address(pci_address, "net")`

also I'd suggest using name `get_interfaces_in_pci` or `get_interfaces_in_pci_address` as we are looking for one or many interfaces...

File: avocado/utils/pci.py [hunk 2]
```diff
@@ -97,20 +98,25 @@ def get_disks_in_pci_address(pci_address):
...
     """
-    Gets network interface(nic) in a PCI address.
+    Gets interface in a PCI address.
 
     :param pci_address: Any segment of a PCI address (1f, 0000:00:1f, ...)
 
-    :return: list of network interfaces in a PCI address.
+    :param clas: Adapter type (FC(fc_host), FCoE(net), NIC(net), SCSI(scsi)..)
```
Comment (ldoktor) [inline avocado/utils/pci.py]: `class`

File: avocado/utils/pci.py [hunk 3]
```diff
@@ -97,20 +98,25 @@ def get_disks_in_pci_address(pci_address):
...
+
```
Comment (ldoktor) [inline avocado/utils/pci.py]: I'd suggest adding one or two examples including returned value to make the usage clearer.

File: avocado/utils/pci.py [hunk 4]
```diff
@@ -135,7 +141,8 @@ def get_slot_from_sysfs(full_pci_address):
 
     :return: slot of PCI address from sysfs.
     """
-    if not os.path.isfile('/sys/bus/pci/devices/%s/devspec' % full_pci_address):
+    if not os.path.isfile('/sys/bus/pci/devices/%s/devspec' %
+                          full_pci_address):
```
Comment (ldoktor) [inline avocado/utils/pci.py]: Please separate this additional change to a different patch as it is not related to the original change. Also (it's a matter of taste) but I personally prefer putting the `%` on the same line with arguments as the argument belongs to `%`, anyway there are different styles and this is currently not enforced).

File: avocado/utils/pci.py [hunk 5]
```diff
@@ -97,20 +98,25 @@ def get_disks_in_pci_address(pci_address):
...
     """
-    iface_path = "/sys/class/net/"
-    net_interfaces_list = []
-    for iface in os.listdir(iface_path):
-        if pci_address in os.readlink("%s%s" % (iface_path, iface)):
-            net_interfaces_list.append(iface)
-    return net_interfaces_list
+    pci_class_path = "/sys/class/%s/" % pci_class
+    if not os.path.isdir(pci_class_path):
+        logging.debug("please pass valid class name")
```
Comment (ldoktor) [inline avocado/utils/pci.py]: Actually this is a failure, isn't it? I'd rather see `assert os.path.isdir(pci_class_path), "Incorrect class name, valid ones are %s" % os.path.listdir("/sys/class")` or raising a `ValueError` (the later is better as `ValueError` better describes the failure)

File: avocado/utils/pci.py [hunk 6]
```diff
@@ -97,20 +98,25 @@ def get_disks_in_pci_address(pci_address):
...
+        return []
+    device_interface_list = []
+    for interface in os.listdir(pci_class_path):
+        if pci_address in os.readlink("%s/%s" % (pci_class_path, interface)):
```
Comment (ldoktor) [inline avocado/utils/pci.py]: please use `os.path.join(pci_class_path, interface)`

File: avocado/utils/pci.py [hunk 7]
```diff
@@ -97,20 +98,25 @@ def get_disks_in_pci_address(pci_address):
...
+            device_interface_list.append(interface)
```
Comment (ldoktor) [inline avocado/utils/pci.py]: actually how about:

```python
return [interface for interface in os.listdir(pci_class_path)
        if pci_address in os.readlink(os.path.join(pci_class_path, interface))]
```

File: avocado/utils/pci.py (hunk 4 above)
Comment (vrbagalkote) [inline avocado/utils/pci.py]: I am seeing below pep8 error so I am forced to make this change in this patch.

pci.py:158:80: E501 line too long (80 > 79 characters)

Comment (ldoktor) [inline avocado/utils/pci.py]: We do not enforce the 80 chars (even though we do our best to keep it), so it won't fail in the CI. Anyway if you want to keep your check happy you can reorganize the commits so first you fix the style fixes and then you add this commit.

File: avocado/utils/pci.py [hunk 8]
```diff
@@ -27,6 +27,8 @@
 from .genio import read_file
 from . import process
 
+log = logging.getLogger('avocado.test')
```
Comment (ldoktor) [inline avocado/utils/pci.py]: You don't use the `log` anywhere, also the use of `avocado.test` directly from `utils` can be visible in other modules, but was declared a bad habit and new code using knowledge of `avocado` in `avocado.utils` should not be accepted...

File: avocado/utils/pci.py [hunk 9]
```diff
@@ -98,25 +100,37 @@ def get_disks_in_pci_address(pci_address):
...
     :param pci_address: Any segment of a PCI address (1f, 0000:00:1f, ...)
 
-    :param clas: Adapter type (FC(fc_host), FCoE(net), NIC(net), SCSI(scsi)..)
+    :param class: Adapter type (FC(fc_host), FCoE(net), NIC(net), SCSI(scsi)..)
 
     :return: list of generic interfaces in a PCI address.
+    e.g: host = pci.get_interfaces_in_pci_address("0001:01:00.0", "net")
+         ['enP1p1s0f0']
+         host = pci.get_interfaces_in_pci_address("0004:01:00.0", "fc_host")
+         ['host6']
```
Comment (ldoktor) [inline avocado/utils/pci.py]: nice

File: avocado/utils/pci.py [hunk 10]
```diff
@@ -98,25 +100,37 @@ def get_disks_in_pci_address(pci_address):
...
-        return []
-    device_interface_list = []
-    for interface in os.listdir(pci_class_path):
-        if pci_address in os.readlink("%s/%s" % (pci_class_path, interface)):
-            device_interface_list.append(interface)
-    return device_interface_list
+        raise ValueError("Please provide valid class name")
+    return [interface for interface in os.listdir(pci_class_path)
+            if pci_address in os.readlink(os.path.join(pci_class_path,
+                                          interface))]
```
Comment (ldoktor) [inline avocado/utils/pci.py]: this is improperly aligned, isn't it?

**Comment by ldoktor** [comment]
Would you please merge the commits, fix the commit typo (Enchanced => Enhance) but I'd suggest different heading `utils: Add "get_interface_in_pci_address" method to "pci.py"` (or similar) as now it's not updating it but adding a new (generic) function.

Last but not least please separate the pep8 fix, you can put the fix as a first commit and then you check should be happy.

Some details about git (ignore it if you already know the drill and sorry for bothering you with that):

```
git fetch --all   # fetches info of all repos
git rebase -i origin/master  # interactively rebase current branch on top of origin/master (replace "origin" with upstream, avocado or whatever remote repo points to the upstream avocado git)
git rebase --continue  # continue rebasing
git citool  # when git-gui installed shows a GUI to select which parts of a commit should be added, useful for splitting commits
git cola  # another GUI frontend for git which allows interactive rebases and other things.
git rebase --ignore-date origin/master  # rebases on top of origin/master ignoring the commit date, useful when you re-order commits and newer commits (by date) are put ahead of older commits (which confuses github)
```

**Comment by vrbagalkote** [comment]
@ldoktor , thanks for your time and patience. I have raised a new PR1910. So I am closing this.
